engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -1119,147 +1288,212 @@ public Class<?> visit(ArrayAccessExpr n, VisitArgs printer) {
-            op = BinaryExpr.Operator.EQUALS;
-        }
+            // Update the AST -- create an EQUALS BinaryExpr, then invert its result.
+            final BinaryExpr equalsExpr = new BinaryExpr(n.getLeft(), n.getRight(), BinaryExpr.Operator.EQUALS);
 
-        boolean isArray = lhType.isArray() || rhType.isArray() || isTypedVector(lhType) || isTypedVector(rhType);
+            final UnaryExpr newExpr = new UnaryExpr(equalsExpr, UnaryExpr.Operator.LOGICAL_COMPLEMENT);
+            newExpr.setComment(QueryLanguageParserComment.BINARY_OP_NEQ_CONVERSION_FLAG);
+            replaceChildExpression(origParent, n, newExpr);
+
+
+            assertNothingPrinted.run();
+            return newExpr.accept(this, printer);
+        }
 
         if (op == BinaryExpr.Operator.EQUALS) {
+            // TODO: it's bit weird to use isNull() here as it's not part of QueryLanguageFunctionUtils.
             if (leftExpr.isNullLiteralExpr()) {
                 if (printer.hasStringBuilder()) {
-                    printer.append("isNull(");
-                    rightExpr.accept(this, printer);
-                    printer.append(")");
+                    final MethodCallExpr isNullExpr = new MethodCallExpr("isNull", rightExpr);
+                    replaceChildExpression(origParent, n, isNullExpr);
+                    return isNullExpr.accept(this, printer);
                 }
                 return boolean.class;
             }
             if (rightExpr.isNullLiteralExpr()) {
                 if (printer.hasStringBuilder()) {
-                    printer.append("isNull(");
-                    leftExpr.accept(this, printer);
-                    printer.append(")");
+                    final MethodCallExpr isNullExpr = new MethodCallExpr("isNull", leftExpr);
+                    replaceChildExpression(origParent, n, isNullExpr);
+                    return isNullExpr.accept(this, printer);
                 }
                 return boolean.class;
             }
         }
 
+        boolean isArray = lhType.isArray() || rhType.isArray() || isTypedVector(lhType) || isTypedVector(rhType);
+
         String methodName = getOperatorName(op) + (isArray ? "Array" : "");
 
         if (printer.hasStringBuilder()) {
-            new MethodCallExpr(null, methodName, new NodeList<>(n.getLeft(), n.getRight()))
-                    .accept(this, printer);
-        }
+            final MethodCallExpr binaryOpOverloadMethod = new MethodCallExpr(methodName, n.getLeft(), n.getRight());
+            replaceChildExpression(origParent, n, binaryOpOverloadMethod); // Replace the BinaryExpr with a method call,
+                                                                           // e.g. "x==y" --> "eq(x, y)"
 
-        // printer.append(methodName + '(');
-        // n.getLeft().accept(this, printer);
-        // printer.append(',');
-        // n.getRight().accept(this, printer);
-        // printer.append(')');
+            assertNothingPrinted.run();
```

Comment:
this assert doesn't make sense to add here, but not below - nothing in this if block could write to the printer (before this line), right? if we have to check here, then it was something before this if check, and should either move the assert before the if (so it validates the returns after the if), or also add more asserts below?

Reply:
below, the printer doesn't have a string builder, so it wouldn't really matter

Reply:
Sure, but it also can't have been appended to in this block until now, right?